clients/unshaded/src/main/java/tachyon/client/file/FileOutStream.java

```diff
@@ -97,8 +101,23 @@ public FileOutStream(long fileId, OutStreamOptions options) throws IOException {
     }
     mClosed = false;
     mCanceled = false;
-    mHostname = options.getHostname();
     mShouldCacheCurrentBlock = mTachyonStorageType.isStore();
+
+    if (options.getLocationPolicy() == null) {
```

Comment:
We should probably have a preconditions in the options creator to reject `null`?